lib/vector-buffers/src/variants/disk_v2/writer.rs

```diff
@@ -133,6 +133,29 @@ where
     EmptyRecord,
 }
 
+impl<T> WriterError<T>
+where
+    T: Bufferable,
+{
+    /// Whether or not this error can be recovered from.
+    ///
+    /// Recovery may be specific to the caller or context. For example, a record that is too large
+    /// to be encoded/serialized could potentially be handled in another codepath, and a data file
+    /// being full be generally be handled by writing to the next data file. However, we can't
+    /// recover from an I/O error, and so on.
+    ///
+    /// TODO: Encoding is currently an operation that consumes the event, due to the constraints of
+    /// the `prost::Message` trait, and by extension, the `Encodable` trait. Similarly, other "event
+    /// has nonsensicial data" errors do not return the event, so _right now_, nothing is recoverable.
+    ///
+    /// In the future, we could likely rework some of the errors to allow passing the event back,
+    /// especially when we have sink error handling and could actually _do_ something with it.
+    #[allow(clippy::unused_self)]
+    pub fn is_recoverable(&self) -> bool {
+        false
+    }
```

Comment:
This looks to be the only implementer of `fn is_recoverable`, and it returns a constant. The comment says we _could_ use this, but do we know we _will_?

Reply:
My intent is that we'll use this to sift out errors that occur during event encoding, such that we could return the event to be passed back via the future sink error handling mechanism.

I mostly just wanted to avoid forgetting, in the future, to _not_ emit an error message if we encountered a recoverable error.